src/bootloaders/shim-systemd.c

```diff
@@ -110,7 +111,7 @@ static char *shim_systemd_get_kernel_dst(const BootManager *manager) {
 
 static bool shim_systemd_install_kernel(const BootManager *manager, const Kernel *kernel) {
         fprintf(stderr, "Call %s\n", __func__);
-        return true;
+        return sd_class_install_kernel_impl(manager, kernel, shim_systemd_get_kernel_dst, NULL);
```

Comment:
Yes, people who will not have Secure Boot enabled will be using the exact same EFI binaries (even though they will be signed) and the exact same boot path.

I'll make sure the new bootloader looks up and cleans up after the previous booting schemes.